src/main/java/org/apache/fineract/api/BatchApi.java

```diff
@@ -66,27 +68,28 @@ public Page<Batch> getBatches(@RequestParam(value = "page", required = false, de
     }
 
     @GetMapping("/batch")
-    public BatchDTO batchDetails(@RequestParam(value = "batchId", required = false) String batchId,
-                                 @RequestParam(value = "requestId", required = false) String requestId) {
+    public ResponseEntity<Object> batchDetails(@RequestParam(value = "batchId", required = false) String batchId,
+                                               @RequestParam(value = "requestId", required = false) String requestId) {
         Batch batch = batchRepository.findByBatchId(batchId);
-        if (batch != null) {
-            if (batch.getResultGeneratedAt() != null) {
-//                Checks if last status was checked before 10 mins
-                if (new Date().getTime() - batch.getResultGeneratedAt().getTime() < 600000) {
-                    return generateDetails(batch);
-                } else {
-                    return generateDetails(batch);
-                }
-            } else {
-                return generateDetails(batch);
-            }
-        } else {
-           Batch batch1 = new Batch();
-           batch1.setBatchId(batchId);
-           batch1.setRequestId(requestId);
-           return generateDetails(batch1);
+
+        if (batch==null) {
+            String errorMessage = "Batch corresponding to batchId: " + batchId + " does not exist.";
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(errorMessage);
         }
+        return ResponseEntity.ok(generateBatchSummaryResponse(batch));
+    }
+
+    @GetMapping("/batch/{batchId}")
+    public ResponseEntity<Object> batchAggregation(@PathVariable(value = "batchId") String batchId,
+                                     @RequestParam(value = "requestId", required = false) String requestId,
+                                     @RequestParam(value = "command", required = false, defaultValue = "aggregate") String command) {
+        Batch batch = batchRepository.findByBatchId(batchId);
 
+        if (batch != null && batch.getResultGeneratedAt() != null
+                && new Date().getTime() - batch.getResultGeneratedAt().getTime() < 600000) {
```

Comment:
1. use brackets to improve the readability of the expression. (Date().getTime() - batch.getResultGeneratedAt().getTime())< 600000

2. why are we checking whether these dates are within 10 minutes?
3.

Reply:
@fynmanoj this was an existing code, but if condition was not serving any purpose. I have removed it now.